### ISSUE TITLE
Use just one set of repositories (give up cksource remote)

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-design#77]
I would consider using just one set of repositories for CKEditor 5. In [CKE 4](github.com/ckeditor/ckeditor-dev) we use the main one and the [cksource](github.com/cksource/ckeditor-dev) remote. This:

* is confusing for some people,
* is sometimes confusing for us (where did I push that branch?!),
* requires more clicking to go to the correct repo (e.g. when you want to make a PR, you need to go switch to the other repo),
* makes less noise in the main repo, what hides our activity),
* requires more work generally,
* in CKE5 we will have waaaay more repositories than in CKE4, so in total there will be a lot of managing.

The reason why we had two remotes is that we wanted to show that CKSource is just one of the contributors. That it’s an open source project that belongs to the community, not only to CKSource. I think the split helps a bit in creating such image, but it doesn’t seem to be super crucial thing and at the same time it has a few downsides.

**Comment by fredck** [comment]
I was checking the situation with ckeditor-dev:

 * The "CKEditor" organization has a clean and easy to understand situation with [only 11 branches](https://github.com/ckeditor/ckeditor-dev/branches/all).
 * The "CKSource" organization, while following a precise standard, has the [massive amount of 277 branches](https://github.com/cksource/ckeditor-dev/branches/all).

If we would have both merged, it would be a mess and most people would just really care about master, only. Ofc, we could have documentation and the such, but we know how things work for real.

Based on the above, this would be a -1 from me.

In the other hand, I understand that this would make things simpler for us, CKSource, as we would be able to avoid having to fork and configure every single repo we'll have in v5.

Based on the above, this would be a +1 from me.

So, to sum up, I'm neutral and I leave this decision for others. Just please take both sides of the coin in consideration.

**Comment by Reinmar** [comment]
> The "CKSource" organization, while following a precise standard, has the massive amount of 277 branches.

That should be cleaned up. And it's caused mostly by the fact that we work with Trac (on GH things are easier) and by the very long history and hundreds of tickets that we've made.

**Comment by Reinmar** [comment]
And one more thing :D Having just one repo will help us taking care of this because it will force us to look at that long list. Right now no one check what's in the cksource/ckeditor-dev repo.

**Comment by fredck** [comment]
Sincerely, I don't feel that this (both two comments above) will change much.

**Comment by wwalc** [comment]
As an outsider just poking around the CKEditor source code, I'd never think about having a look into cksource/ckeditor-dev repos. For me it's natural that when I look at some project I expect to find (any) code there directly in the original repository.

**Comment by Reinmar** [comment]
And one more thing :D In CKE5 we will have *a lot* more repos in general, so there will be less tickets per repo, so there will be less branches per repo any way.

**Comment by pjasiun** [comment]
In fact we need some tools to handle working with repos and npm, so there is no a big difference if there is one repo or two. The tool will handle it anyway.

**Comment by pjasiun** [comment]
But in fact I agree with most of @Reinmar points. I also do not think having 2 repos very useful, and there are small irritating details, like the fact you can not easily switch from master to the ticket branch if you are browsing it on the GitHub.

**Comment by oleq** [comment]
I've never seen a project using the same architecture that we have in v4 on GitHub, I mean, two remotes to separate "clean" and "dirty" side of the project. That's odd. I got used to it but it is sort of... confusing. And redundant. If we expect people to contribute and understand CKEditor, we should not obscure the code and our workflow.

So :+1:

**Comment by Reinmar** [comment]
OK, so since most of us don't see big problems in keeping most of the work in a single repo, I think we can call it a decision.

I have one more idea though, that we can keep the `cksource/*` remotes for more R&D kind of work when we don't have a well defined ticket (goal) yet. And we can have them only for some specific high traffic repositories like ckeditor5-core. WDYT?

**Comment by pjasiun** [comment]
:+1:

**Comment by scofalik** [comment]
:+1:

**Comment by Reinmar** [comment]
We started using just one set of remotes and it works perfectly fine for us. I guess the topic does not need any other summary.
